lua/treesitter-context.lua

```diff
@@ -621,6 +633,18 @@ local function get_relative_line_num(ctx_node_line_num)
   return cursor_line_num - ctx_node_line_num - num_folded_lines
 end
 
+local function horizontal_scroll_contexts()
+  if context_winid == nil then
+    return
+  end
+  local current_win_view = vim.api.nvim_win_call(context_winid, function ()
+    return vim.fn.winsaveview()
```

Comment:
I think this would be simpler if you just called `winsaveview` for the main window and the context window and see if they are different. If they are, you can then call `winrestview({leftcol = ...})`.

Reply:
`winsaveview` returns more information than just `leftcol`. I can build in a check and only call `winrestview` on change, but since the function is now called on horizontal scroll only, this only avoids a single call every window change.

Reply:
> winsaveview returns more information than just leftcol.

That shouldn't be a problem. `winsaveview()` is actually much cheaper than I thought as it just returns internal variables:

```c
/// "winsaveview()" function
static void f_winsaveview(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
{
  tv_dict_alloc_ret(rettv);
  dict_T *dict = rettv->vval.v_dict;

  tv_dict_add_nr(dict, S_LEN("lnum"), (varnumber_T)curwin->w_cursor.lnum);
  tv_dict_add_nr(dict, S_LEN("col"), (varnumber_T)curwin->w_cursor.col);
  tv_dict_add_nr(dict, S_LEN("coladd"), (varnumber_T)curwin->w_cursor.coladd);
  update_curswant();
  tv_dict_add_nr(dict, S_LEN("curswant"), (varnumber_T)curwin->w_curswant);

  tv_dict_add_nr(dict, S_LEN("topline"), (varnumber_T)curwin->w_topline);
  tv_dict_add_nr(dict, S_LEN("topfill"), (varnumber_T)curwin->w_topfill);
  tv_dict_add_nr(dict, S_LEN("leftcol"), (varnumber_T)curwin->w_leftcol);
  tv_dict_add_nr(dict, S_LEN("skipcol"), (varnumber_T)curwin->w_skipcol);
}
```

It is `winrestview()` which does a bunch of stuff.

Reply:
If i only call `winrestview` if the scroll is different, should i move it back into the update function then? I feel like that keeps the code a whole lot simpler.

Reply:
Ok

Reply:
Done